### PR TITLE
Test: Add DNS proxy test

### PR DESCRIPTION
**Pull request by eloycoto**
Two new tests: 

- Added a new test to make sure that the old TOCIDR are not deleted when a
new FQDN rule is added.
- Added a new test that uses DNS roundrobin to validate that the FQDN is working correctly. 

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6298)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/fqdn.go [hunk 1]
```diff
@@ -245,4 +304,67 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
...
+					"rules": {
+						"dns": [
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "cilium.test"
```
Comment (tgraf) [inline test/runtime/fqdn.go]: This is whitelisting `cilium.test` but we test to `world1.cilium.test`

**Review by tgraf — CHANGES_REQUESTED** [review]
These tests look great! All tests seem to whitelist `cilium.test` but then test connectivity to `world.cilium.test` or similar. Shouldn't this break?

File: test/runtime/fqdn.go (hunk 1 above)
Comment (eloycoto) [reply test/runtime/fqdn.go]: sorry, fix this in the next commit: 
https://github.com/cilium/cilium/pull/6298/commits/fcfd2fe5c97be1d0669f4b80da06326cb7109223

Will fix now.

**Comment by eloycoto** [comment]
test-me-please (Download timeout: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8261/)

**Review by tgraf — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**